lib/core/src/Cardano/Wallet/DB/Sqlite/Types.hs

```diff
@@ -497,13 +498,23 @@ instance PersistFieldSql ScriptHash where
 -- KeyHash
 
 instance ToText KeyHash where
-    toText (KeyHash sh) =
-        T.decodeUtf8 $ convertToBase Base16 sh
+    toText (KeyHash keyRole sh) =
+        T.append keyRoleTxt $ T.decodeUtf8 $ convertToBase Base16 sh
+      where
+          keyRoleTxt = case keyRole of
+              CA.Payment -> "0"
+              CA.Delegation -> "2"
 
 instance FromText KeyHash where
-    fromText = bimap textDecodingError KeyHash
-        . convertFromBase Base16
-        . T.encodeUtf8
+    fromText txt = do
+        (keyRole, txt') <- case T.uncons txt of
+            Just (firstChar, rest) ->case firstChar of
+                '0' -> pure (CA.Payment, rest)
+                '2' -> pure (CA.Delegation, rest)
+                _ -> Left $ TextDecodingError "KeyHash should begin with either '0' or '1'."
+            Nothing -> Left $ TextDecodingError "KeyHash should not be empty."
+        bimap textDecodingError (KeyHash keyRole)
+            $ convertFromBase Base16 $ T.encodeUtf8 txt'
```

Comment:
This is a red-herring to me :thinking: ... 

Why would you need to differentiate them in the database? Surely, the context when deserializing from the database is enough to assign the right role to the key hash. If it's not, then it should likely go in a separate column, instead of being serialized all at once.

Regardless, this will require a database migration and I fail to see how existing databases can be migrated :thinking: ? (or said differently, why is this distinction only occurs now?)

Reply:
I checked that KeyHash is not stored. These instances are not needed in the end. so removed them. They are some remnants of past version of shared wallet I forgot to erase.